stock_warehouse_orderpoint_stock_info/models/stock_warehouse_orderpoint.py

```diff
@@ -7,21 +7,27 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 from odoo import api, fields, models
+from collections import defaultdict
 
 
 class StockWarehouseOrderpoint(models.Model):
     _inherit = 'stock.warehouse.orderpoint'
 
     @api.multi
     def _compute_product_available_qty(self):
-        for rec in self:
-            product_available = rec.product_id.with_context(
-                location=rec.location_id.id
-                )._product_available()[rec.product_id.id]
-            rec.product_location_qty = product_available['qty_available']
-            rec.incoming_location_qty = product_available['incoming_qty']
-            rec.outgoing_location_qty = product_available['outgoing_qty']
-            rec.virtual_location_qty = product_available['virtual_available']
+        op_by_loc = defaultdict(set)
+        for order in self:
+            op_by_loc[order.location_id].add(order)
```

Comment:
In a recordset, you add data by using the `|=` operator.
`op_by_loc[order.location_id] |= order`

Note: The `+=` operator also works, but sometimes adds duplicates in the recordset, that's why I prefer using `|=`.

Reply:
@sylvain-garancher Good catch. Indeed I hadn't seen the call to `mapped` at line 22